app/src/main/java/com/example/wnews/UserProvider.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.wnews
+
+import android.content.SharedPreferences
+import com.example.wnews.models.UserAuth
+import com.google.gson.Gson
+
+object UserProvider {
+
+    lateinit var userAuth: UserAuth
+        private set
+
+
+    fun isLoggedIn(): Boolean {
+        return userAuth.userAuthId != 0
+
+    }
+
+    fun intanceUserAuth(sharedPreferences: SharedPreferences) {
+
+        val sharedPrefString = sharedPreferences.getString("UserAuth", "")
```

Comment:
Extract `UserAuth` string into a constant, so that we can be sure that no typo will cause hard to spot bugs. This is specially useful when the same string needs to be accessed from several files/methods, like in this case.